internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -507,8 +509,8 @@ func (r *PreprovisioningImageReconciler) getIronicConfig(ctx context.Context, lo
 		return nil, fmt.Errorf("Failed to determine ironic config")
 	}
 
-	if iccConfig.IronicBaseURL == "" && iccConfig.IronicInspectorBaseUrl == "" {
-		err := r.fillIronicServiceURLs(ctx, infraEnv, infraEnvInternal, iccConfig)
+	if iccConfig.IronicBaseURL == "" {
+		err := r.fillIronicBaseURL(ctx, infraEnv, infraEnvInternal, iccConfig)
```

Comment:
I think the problem here is that we could still get into a situation where the config is not pulled from the BMO configmap, we get the ironic image from the hub then fail to fill in the inspector URL (even though we should in that case).